openstudiocore/src/energyplus/ForwardTranslator/ForwardTranslateElectricLoadCenterDistribution.cpp

```diff
@@ -88,38 +80,222 @@ boost::optional<IdfObject> ForwardTranslator::translateElectricLoadCenterDistrib
   //  }
   //}
 
-  for (auto& generator : modelObject.generators()){
+  for (auto& generator : modelObject.generators()) {
     boost::optional<IdfObject> generatorIdf = translateAndMapModelObject(generator);
 
-    if (generatorIdf){
+    if (generatorIdf) {
       IdfExtensibleGroup generatorGroup = generatorsIdf.pushExtensibleGroup();
 
       generatorGroup.setString(ElectricLoadCenter_GeneratorsExtensibleFields::GeneratorName, generatorIdf->name().get());
-      
+
       generatorGroup.setString(ElectricLoadCenter_GeneratorsExtensibleFields::GeneratorObjectType, generator.generatorObjectType());
-      
-      d = generator.ratedElectricPowerOutput();
-      if (d){
-        generatorGroup.setDouble(ElectricLoadCenter_GeneratorsExtensibleFields::GeneratorRatedElectricPowerOutput, *d);
+
+      optD = generator.ratedElectricPowerOutput();
+      if (optD) {
+        generatorGroup.setDouble(ElectricLoadCenter_GeneratorsExtensibleFields::GeneratorRatedElectricPowerOutput, *optD);
       }
-      
+
       schedule = generator.availabilitySchedule();
-      if (schedule){
+      if (schedule) {
         boost::optional<IdfObject> scheduleIdf = translateAndMapModelObject(*schedule);
-        if (scheduleIdf){
+        if (scheduleIdf) {
           generatorGroup.setString(ElectricLoadCenter_GeneratorsExtensibleFields::GeneratorAvailabilityScheduleName, scheduleIdf->name().get());
         }
       }
 
-      d = generator.ratedThermalToElectricalPowerRatio();
-      if (d){
-        generatorGroup.setDouble(ElectricLoadCenter_GeneratorsExtensibleFields::GeneratorRatedThermaltoElectricalPowerRatio, *d);
+      optD = generator.ratedThermalToElectricalPowerRatio();
+      if (optD) {
+        generatorGroup.setDouble(ElectricLoadCenter_GeneratorsExtensibleFields::GeneratorRatedThermaltoElectricalPowerRatio, *optD);
       }
-    } else{
+    } else {
       LOG(Warn, "Could not translate generator '" << generator.name().get() << "' on ElectricLoadCenter:Distribution '" << idfObject.name().get() << "'")
     }
   }
 
+  // NEW
+
+  // Logic based on Electrical Buss Type to translate or not translate inverters, storage
+  std::string bussType = modelObject.electricalBussType();
+ 
+  // Translate the buss Type already
+  idfObject.setString(ElectricLoadCenter_DistributionFields::ElectricalBussType, bussType);
+  
+  /// Inverter and Buss Type
+  boost::optional<Inverter> inverter = modelObject.inverter();
```

Comment:
My thought would be that setting an inverter should trump what the bus type is.  I also had the idea that we should detect AC or DC by the generators on the list.  The model should prevent you from mixing AC and DC generators.